packages/core-interop/src/test/Metadata.test.ts

```diff
@@ -63,6 +63,39 @@ describe("createMetadataProvider", () => {
       expect(typeof schema.getClass === "function").to.be.true;
     });
 
+    it("caches schema request", async () => {
+      const matchSchemaName = sinon.match((key: SchemaKey) => key.compareByName("x"));
+      const schemaContext = {
+        getSchema: sinon
+          .stub<[SchemaKey], CoreSchema>()
+          .withArgs(matchSchemaName)
+          .resolves({
+            name: "y",
+          } as unknown as CoreSchema),
+      } as unknown as SchemaContext;
+
+      const provider = createMetadataProvider(schemaContext);
+      await provider.getSchema("x");
+      await provider.getSchema("x");
```

Comment:
I'd say this should be `await Promise.all([provider.getSchema("x"), provider.getSchema("x")])`